vmm-reference-gdt.md

```diff
@@ -0,0 +1,185 @@
+# [GDT](../vmm-reference/src/vm-vcpu-ref/src/x86_64/gdt.rs)
+This module provides abstractions for building a Global Descriptor Table (GDT).
+
+## Segment Descriptors
+Segment Descriptors for the GDT have been defined as follows:
+```rs
+pub struct SegmentDescriptor(pub u64);
+```
+
+The following function creates a `SegmentDescriptor` given base, limit and flags.
+The way to convert these numbers to a 64 - bit number has been described [here](https://wiki.osdev.org/Global_Descriptor_Table#Segment_Descriptor). 
+
+`base` is a 32 bit value and specifies the address where this segment starts.
+
+`limit` is a 32 bit value, of which only the least significant 20 bits are used. It tells the maximum addressable unit.
+
+`flags` is a 16 bit value, of which 12 are used. It has two parts. The last 8 bits are referred to as Access Byte. The first 4 bits are some more flags.
+
+```rs
+    pub fn from(flags: u16, base: u32, limit: u32) -> Self;
+```
+<table class="wikitable">
```

Comment:
please change to markdown tables so it is more easily viewed in markdown itself.